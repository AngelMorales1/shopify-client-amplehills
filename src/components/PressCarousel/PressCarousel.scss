@import 'settings';

.PressCarousel {
  &__card {
    border-radius: radius('s');
    margin: spacing(4);
<<<<<<< HEAD
    width: spacing(65);

    @media ($medium-up) {
      width: spacing(75);
=======
    min-width: spacing(65);

    @media ($medium-up) {
      min-width: spacing(75);
>>>>>>> Make HorizontalCarousel base component
    }
  }

  &__logo {
    width: spacing(21);
    height: spacing(10);
    object-fit: contain;
  }

  &__quote {
    font-size: $font-5;
    margin-bottom: spacing(6);

    @media ($medium-up) {
      font-size: $font-6;
      margin-bottom: spacing(16);
    }
  }

  &__button {
    > span {
      margin-bottom: 1px;
    }
  }
}
