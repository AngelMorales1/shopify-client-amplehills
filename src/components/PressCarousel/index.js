import React from 'react';
import PropTypes from 'prop-types';
import get from 'utils/get';
import cx from 'classnames';

import styles from './PressCarousel.scss';
import { Image, Button, HorizontalCarousel } from 'components/base';

const PressCarousel = ({ block, z, pressItems, setRef }) => {
  const fields = get(block, 'fields', {});
  const isDripOn = get(fields, 'drip', false);
  const showCardNumber = get(fields, 'showCardNumber', null);
  const sortByLatest = get(fields, 'sortByLatest', true);
  const pressItemsInBlock = get(fields, 'pressItems.fragments', []);
  const isCustomOrder = !!pressItemsInBlock.length;
  const selectedPressItems = isCustomOrder
    ? pressItemsInBlock
    : get(pressItems, 'fragments', []);
  let selectedPressItemsIds = selectedPressItems.map(fragment =>
    get(fragment[0], 'value', '')
  );

  if (typeof showCardNumber === 'number') {
    selectedPressItemsIds = selectedPressItemsIds.slice(0, showCardNumber);
  }

  return (
    <div
      ref={refBlock => setRef(refBlock)}
      style={{ zIndex: z }}
      className={cx(styles['PressCarousel'], { drip: isDripOn }, 'bg-bees-wax')}
    >
      <HorizontalCarousel
        title={get(fields, 'title', '')}
        buttonLink={get(fields, 'buttonLink', '')}
        buttonLabel={get(fields, 'buttonLabel', '')}
        isReverseOrder={!isCustomOrder && !sortByLatest ? true : false}
      >
        {selectedPressItemsIds.map((pressItemId, i) => {
          const selectedPressItemsSimpleFragments = isCustomOrder
            ? get(fields, 'pressItems.simpleFragments', {})
            : get(pressItems, 'simpleFragments', {});
          const selectedPressItem = get(
            selectedPressItemsSimpleFragments,
            pressItemId,
            {}
          );

          return (
            <div
              key={pressItemId}
              className={cx(
                styles['PressCarousel__card'],
                'bg-white p3 flex flex-column justify-center items-center'
              )}
            >
              <Image
                className={cx(styles['PressCarousel__logo'])}
                src={get(selectedPressItem, 'logoImage.data', '')}
                alt={`${get(selectedPressItem, 'title', '')} logo`}
              />
              <span
                className={cx(
                  styles['PressCarousel__quote'],
                  'carter text-peach center py3'
                )}
              >
                {get(selectedPressItem, 'quote', '')}
              </span>
              <Button
                className={cx(
                  styles['PressCarousel__button'],
                  'uppercase detail'
                )}
                to={get(selectedPressItem, 'linkUrl', '')}
                label="Read about it"
                variant="primary-small"
                color="peach"
              />
            </div>
          );
        })}
      </HorizontalCarousel>
    </div>
  );
};

HorizontalCarousel.propTypes = {
  block: PropTypes.shape({
    fields: PropTypes.shape({
      buttonLabel: PropTypes.string,
      buttonLink: PropTypes.string,
      pressItems: PropTypes.shape({
        fragments: PropTypes.arrayOf(PropTypes.array),
        simpleFragments: PropTypes.object
      }),
      customOrder: PropTypes.bool,
      drip: PropTypes.bool,
      sortByLatest: PropTypes.bool,
      title: PropTypes.string
    }),
    sys: PropTypes.shape({
      id: PropTypes.string
    })
  }),
  z: PropTypes.number,
  pressItems: PropTypes.shape({
    fragments: PropTypes.arrayOf(PropTypes.array),
    simpleFragments: PropTypes.object
  }),
  setRef: PropTypes.func
};

HorizontalCarousel.defaultProps = {
  block: {
    fields: {
      buttonLabel: '',
      buttonLink: '',
      pressItems: {
        fragments: [],
        simpleFragments: {}
      },
      customOrder: false,
      drip: false,
      sortByLatest: true,
      title: ''
    },
    sys: {
      id: ''
    }
  },
  z: 0,
  pressItems: {
    fragments: [],
    simpleFragments: {}
  },
  setRef: () => {}
};

export default PressCarousel;
