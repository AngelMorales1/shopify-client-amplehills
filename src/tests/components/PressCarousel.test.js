import React from 'react';
import { shallow } from 'enzyme';

import PressCarousel from 'components/PressCarousel';

it('renders without data', () => {
  const component = shallow(<PressCarousel />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(
    <PressCarousel
      block={{
        fields: {
          buttonLabel: 'foo',
          buttonLink: 'foo',
          pressItems: {
            fragments: [
              [
                { key: 'foo', value: 'foo' },
                { key: 'foo', value: 'foo' },
                { key: 'foo', value: 'foo' },
                { key: 'foo', value: 'foo' },
                { key: 'foo', value: 'foo' }
              ]
            ],
            simpleFragments: { foo: {} }
          },
          customOrder: false,
          drip: false,
          sortByLatest: true,
          title: 'foo'
        },
        sys: {
          id: 'foo'
        }
      }}
      z={0}
      pressItems={{
        fragments: [
          [
            { key: 'foo', value: 'foo' },
            { key: 'foo', value: 'foo' },
            { key: 'foo', value: 'foo' },
            { key: 'foo', value: 'foo' },
            { key: 'foo', value: 'foo' }
          ]
        ],
        simpleFragments: { foo: {} }
      }}
    />
  );

  expect(component).toMatchSnapshot();
});
