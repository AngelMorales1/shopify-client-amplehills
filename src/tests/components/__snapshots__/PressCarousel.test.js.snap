// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PressCarousel
    block={
        Object {
            "fields": Object {
              "buttonLabel": "foo",
              "buttonLink": "foo",
              "customOrder": false,
              "drip": false,
              "pressItems": Object {
                "fragments": Array [
                  Array [
                    Object {
                      "key": "foo",
                      "value": "foo",
                    },
                    Object {
                      "key": "foo",
                      "value": "foo",
                    },
                    Object {
                      "key": "foo",
                      "value": "foo",
                    },
                    Object {
                      "key": "foo",
                      "value": "foo",
                    },
                    Object {
                      "key": "foo",
                      "value": "foo",
                    },
                  ],
                ],
                "simpleFragments": Object {
                  "foo": Object {},
                },
              },
              "sortByLatest": true,
              "title": "foo",
            },
            "sys": Object {
              "id": "foo",
            },
          }
    }
    pressItems={
        Object {
            "fragments": Array [
              Array [
                Object {
                  "key": "foo",
                  "value": "foo",
                },
                Object {
                  "key": "foo",
                  "value": "foo",
                },
                Object {
                  "key": "foo",
                  "value": "foo",
                },
                Object {
                  "key": "foo",
                  "value": "foo",
                },
                Object {
                  "key": "foo",
                  "value": "foo",
                },
              ],
            ],
            "simpleFragments": Object {
              "foo": Object {},
            },
          }
    }
    z={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <HorizontalCarousel
        block={
                Object {
                        "fields": Object {
                          "buttonLabel": "",
                          "buttonLink": "",
                          "customOrder": false,
                          "drip": false,
                          "pressItems": Object {
                            "fragments": Array [],
                            "simpleFragments": Object {},
                          },
                          "sortByLatest": true,
                          "title": "",
                        },
                        "sys": Object {
                          "id": "",
                        },
                      }
        }
        buttonLabel="foo"
        buttonLink="foo"
        isReverseOrder={false}
        pressItems={
                Object {
                        "fragments": Array [],
                        "simpleFragments": Object {},
                      }
        }
        title="foo"
        z={0}
>
        <div
                className="bg-white p3 flex flex-column justify-center items-center"
        >
                <Image
                        alt=" logo"
                        bg={false}
                        className=""
                        onImgLoad={[Function]}
                        src=""
                        style={Object {}}
                />
                <span
                        className="carter text-peach center py3"
                >
                        ""
                </span>
                <Button
                        className="uppercase detail"
                        color="peach"
                        disabled={false}
                        fullWidth={false}
                        hover="shadow"
                        label="Read about it"
                        onClick={[Function]}
                        shadow={false}
                        to=""
                        type="button"
                        variant="primary-small"
                />
        </div>
</HorizontalCarousel>,
      "className": "bg-bees-wax",
      "style": Object {
        "zIndex": 0,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "block": Object {
          "fields": Object {
            "buttonLabel": "",
            "buttonLink": "",
            "customOrder": false,
            "drip": false,
            "pressItems": Object {
              "fragments": Array [],
              "simpleFragments": Object {},
            },
            "sortByLatest": true,
            "title": "",
          },
          "sys": Object {
            "id": "",
          },
        },
        "buttonLabel": "foo",
        "buttonLink": "foo",
        "children": Array [
          <div
            className="bg-white p3 flex flex-column justify-center items-center"
>
            <Image
                        alt=" logo"
                        bg={false}
                        className=""
                        onImgLoad={[Function]}
                        src=""
                        style={Object {}}
            />
            <span
                        className="carter text-peach center py3"
            >
                        ""
            </span>
            <Button
                        className="uppercase detail"
                        color="peach"
                        disabled={false}
                        fullWidth={false}
                        hover="shadow"
                        label="Read about it"
                        onClick={[Function]}
                        shadow={false}
                        to=""
                        type="button"
                        variant="primary-small"
            />
</div>,
        ],
        "isReverseOrder": false,
        "pressItems": Object {
          "fragments": Array [],
          "simpleFragments": Object {},
        },
        "title": "foo",
        "z": 0,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "foo",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Image
                alt=" logo"
                bg={false}
                className=""
                onImgLoad={[Function]}
                src=""
                style={Object {}}
/>,
              <span
                className="carter text-peach center py3"
>
                ""
</span>,
              <Button
                className="uppercase detail"
                color="peach"
                disabled={false}
                fullWidth={false}
                hover="shadow"
                label="Read about it"
                onClick={[Function]}
                shadow={false}
                to=""
                type="button"
                variant="primary-small"
/>,
            ],
            "className": "bg-white p3 flex flex-column justify-center items-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": " logo",
                "bg": false,
                "children": null,
                "className": "",
                "onImgLoad": [Function],
                "src": "",
                "style": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "\\"\\"",
                "className": "carter text-peach center py3",
              },
              "ref": null,
              "rendered": "\\"\\"",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": null,
                "className": "uppercase detail",
                "color": "peach",
                "disabled": false,
                "fullWidth": false,
                "hover": "shadow",
                "label": "Read about it",
                "onClick": [Function],
                "shadow": false,
                "to": "",
                "type": "button",
                "variant": "primary-small",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <HorizontalCarousel
          block={
                    Object {
                              "fields": Object {
                                "buttonLabel": "",
                                "buttonLink": "",
                                "customOrder": false,
                                "drip": false,
                                "pressItems": Object {
                                  "fragments": Array [],
                                  "simpleFragments": Object {},
                                },
                                "sortByLatest": true,
                                "title": "",
                              },
                              "sys": Object {
                                "id": "",
                              },
                            }
          }
          buttonLabel="foo"
          buttonLink="foo"
          isReverseOrder={false}
          pressItems={
                    Object {
                              "fragments": Array [],
                              "simpleFragments": Object {},
                            }
          }
          title="foo"
          z={0}
>
          <div
                    className="bg-white p3 flex flex-column justify-center items-center"
          >
                    <Image
                              alt=" logo"
                              bg={false}
                              className=""
                              onImgLoad={[Function]}
                              src=""
                              style={Object {}}
                    />
                    <span
                              className="carter text-peach center py3"
                    >
                              ""
                    </span>
                    <Button
                              className="uppercase detail"
                              color="peach"
                              disabled={false}
                              fullWidth={false}
                              hover="shadow"
                              label="Read about it"
                              onClick={[Function]}
                              shadow={false}
                              to=""
                              type="button"
                              variant="primary-small"
                    />
          </div>
</HorizontalCarousel>,
        "className": "bg-bees-wax",
        "style": Object {
          "zIndex": 0,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "block": Object {
            "fields": Object {
              "buttonLabel": "",
              "buttonLink": "",
              "customOrder": false,
              "drip": false,
              "pressItems": Object {
                "fragments": Array [],
                "simpleFragments": Object {},
              },
              "sortByLatest": true,
              "title": "",
            },
            "sys": Object {
              "id": "",
            },
          },
          "buttonLabel": "foo",
          "buttonLink": "foo",
          "children": Array [
            <div
              className="bg-white p3 flex flex-column justify-center items-center"
>
              <Image
                            alt=" logo"
                            bg={false}
                            className=""
                            onImgLoad={[Function]}
                            src=""
                            style={Object {}}
              />
              <span
                            className="carter text-peach center py3"
              >
                            ""
              </span>
              <Button
                            className="uppercase detail"
                            color="peach"
                            disabled={false}
                            fullWidth={false}
                            hover="shadow"
                            label="Read about it"
                            onClick={[Function]}
                            shadow={false}
                            to=""
                            type="button"
                            variant="primary-small"
              />
</div>,
          ],
          "isReverseOrder": false,
          "pressItems": Object {
            "fragments": Array [],
            "simpleFragments": Object {},
          },
          "title": "foo",
          "z": 0,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "foo",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Image
                  alt=" logo"
                  bg={false}
                  className=""
                  onImgLoad={[Function]}
                  src=""
                  style={Object {}}
/>,
                <span
                  className="carter text-peach center py3"
>
                  ""
</span>,
                <Button
                  className="uppercase detail"
                  color="peach"
                  disabled={false}
                  fullWidth={false}
                  hover="shadow"
                  label="Read about it"
                  onClick={[Function]}
                  shadow={false}
                  to=""
                  type="button"
                  variant="primary-small"
/>,
              ],
              "className": "bg-white p3 flex flex-column justify-center items-center",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "alt": " logo",
                  "bg": false,
                  "children": null,
                  "className": "",
                  "onImgLoad": [Function],
                  "src": "",
                  "style": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "\\"\\"",
                  "className": "carter text-peach center py3",
                },
                "ref": null,
                "rendered": "\\"\\"",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": null,
                  "className": "uppercase detail",
                  "color": "peach",
                  "disabled": false,
                  "fullWidth": false,
                  "hover": "shadow",
                  "label": "Read about it",
                  "onClick": [Function],
                  "shadow": false,
                  "to": "",
                  "type": "button",
                  "variant": "primary-small",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PressCarousel />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <HorizontalCarousel
        block={
                Object {
                        "fields": Object {
                          "buttonLabel": "",
                          "buttonLink": "",
                          "customOrder": false,
                          "drip": false,
                          "pressItems": Object {
                            "fragments": Array [],
                            "simpleFragments": Object {},
                          },
                          "sortByLatest": true,
                          "title": "",
                        },
                        "sys": Object {
                          "id": "",
                        },
                      }
        }
        buttonLabel=""
        buttonLink=""
        isReverseOrder={false}
        pressItems={
                Object {
                        "fragments": Array [],
                        "simpleFragments": Object {},
                      }
        }
        title=""
        z={0}
>
        
</HorizontalCarousel>,
      "className": "bg-bees-wax",
      "style": Object {
        "zIndex": undefined,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "block": Object {
          "fields": Object {
            "buttonLabel": "",
            "buttonLink": "",
            "customOrder": false,
            "drip": false,
            "pressItems": Object {
              "fragments": Array [],
              "simpleFragments": Object {},
            },
            "sortByLatest": true,
            "title": "",
          },
          "sys": Object {
            "id": "",
          },
        },
        "buttonLabel": "",
        "buttonLink": "",
        "children": Array [],
        "isReverseOrder": false,
        "pressItems": Object {
          "fragments": Array [],
          "simpleFragments": Object {},
        },
        "title": "",
        "z": 0,
      },
      "ref": null,
      "rendered": Array [],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <HorizontalCarousel
          block={
                    Object {
                              "fields": Object {
                                "buttonLabel": "",
                                "buttonLink": "",
                                "customOrder": false,
                                "drip": false,
                                "pressItems": Object {
                                  "fragments": Array [],
                                  "simpleFragments": Object {},
                                },
                                "sortByLatest": true,
                                "title": "",
                              },
                              "sys": Object {
                                "id": "",
                              },
                            }
          }
          buttonLabel=""
          buttonLink=""
          isReverseOrder={false}
          pressItems={
                    Object {
                              "fragments": Array [],
                              "simpleFragments": Object {},
                            }
          }
          title=""
          z={0}
>
          
</HorizontalCarousel>,
        "className": "bg-bees-wax",
        "style": Object {
          "zIndex": undefined,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "block": Object {
            "fields": Object {
              "buttonLabel": "",
              "buttonLink": "",
              "customOrder": false,
              "drip": false,
              "pressItems": Object {
                "fragments": Array [],
                "simpleFragments": Object {},
              },
              "sortByLatest": true,
              "title": "",
            },
            "sys": Object {
              "id": "",
            },
          },
          "buttonLabel": "",
          "buttonLink": "",
          "children": Array [],
          "isReverseOrder": false,
          "pressItems": Object {
            "fragments": Array [],
            "simpleFragments": Object {},
          },
          "title": "",
          "z": 0,
        },
        "ref": null,
        "rendered": Array [],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
